File: Example/SDKDemo/FirstViewController.m
//
//  FirstViewController.m
//  SDKDemo
//
//  Created by Paul Sauer on 5/19/20.
//  Copyright © 2020 QuadPay. All rights reserved.
//

#import "FirstViewController.h"
#import <ZipSDK/QuadPaySDK.h>

@interface FirstViewController () <QuadPayCheckoutDelegate>

@end

@implementation FirstViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    
}

- (IBAction)checkoutButtonPressed:(id)sender {
    /*
        This action handler is where the QuadPay checkout is started
     */

    QuadPayCheckoutDetails* details = [QuadPayCheckoutDetails alloc];
    details.amount = [NSDecimalNumber decimalNumberWithString:@"94.40" locale:NULL];
    details.customerPhoneNumber = @"+8146225937";
    details.customerCity = @"New York";
    details.customerState = @"NY";
    details.customerAddressLine1 = @"240 Meeker Ave";
    details.customerAddressLine2 = @"Apt 35";
    details.customerPostalCode = @"11211";
    details.customerCountry = @"US";
    details.customerFirstName = @"Quincy";
    details.customerLastName = @"Payman";
    details.customerEmail = @"sdk_example@quadpay.com";
    details.checkoutFlow = @"express";

    QuadPayCheckoutViewController* view = [QuadPayCheckoutViewController startCheckout:self details:details];
    [self presentViewController:view animated:YES completion:nil];
}

- (void)didFailWithError:(QuadPayCheckoutViewController*)viewController error:(nonnull NSString *)error {
    [viewController dismissViewControllerAnimated:true completion:^ {}];
    NSLog(@"%@", [NSString stringWithFormat:@"QuadPay checkout encountered an error %@", error]);
}

- (void)checkoutCancelled:(QuadPayCheckoutViewController*)viewController reason:(NSString *)reason {
    NSLog(@"%@", [NSString stringWithFormat:@"User cancelled QuadPay checkout with reason %@", reason]);
    [viewController dismissViewControllerAnimated:true completion:^ {
        /*
            Your code to handle cancellation (if any), this demo just shows an alert!
         */
        [self showUserCancelledAlert];
    }];
}

- (void) checkoutSuccessful:(QuadPayCheckoutViewController*)viewController orderId:(NSString*)orderId {
    NSLog(@"%@", [NSString stringWithFormat:@"Confirmation orderId %@", orderId]);
    [viewController dismissViewControllerAnimated:true completion:^ {
        /*
            Pass this token to your backend to complete the order!
         */
        [self showCheckoutSuccessAlert];
    }];
}

- (void) checkoutSuccessful:(QuadPayCheckoutViewController*)viewController orderId:(NSString*)orderId customer:(QuadPayCustomer*)customer {
    NSLog(@"%@", [NSString stringWithFormat:@"Confirmation orderId %@, customer: %@", orderId, [customer toString]]);
    [viewController dismissViewControllerAnimated:true completion:^ {
        /*
         Pass this token to your backend to complete the order!
         */
        [self showCheckoutSuccessAlert];
    }];
}


- (void)showUserCancelledAlert {
    UIAlertController* alert = [UIAlertController alertControllerWithTitle:@"User Cancelled"
                                                                   message:@"The user has cancelled the QuadPay checkout!"
                                                            preferredStyle:UIAlertControllerStyleAlert];
    
    UIAlertAction* defaultAction = [UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault
                                                          handler:^(UIAlertAction * action) {}];
    
    [alert addAction:defaultAction];
    [self presentViewController:alert animated:YES completion:nil];
}

- (void)showCheckoutSuccessAlert {
    UIAlertController* alert = [UIAlertController alertControllerWithTitle:@"Checkout Succeeded"
                                                                   message: @"QuadPay checkout succeeded"
                                                            preferredStyle:UIAlertControllerStyleAlert];
    
    UIAlertAction* defaultAction = [UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault
                                                          handler:^(UIAlertAction * action) {}];
    
    [alert addAction:defaultAction];
    [self presentViewController:alert animated:YES completion:nil];
}

@end

File: SwiftExample/SDKTestSwift/FirstViewController.swift
//
//  FirstViewController.swift
//  SDKTestSwift
//
//  Created by Paul Sauer on 6/4/20.
//  Copyright © 2020 QuadPay. All rights reserved.
//

import UIKit
import QuadPaySDK

class FirstViewController: UIViewController, QuadPayCheckoutDelegate {
    func didFailWithError(_ viewController: QuadPayCheckoutViewController, error: String) {
        viewController.dismiss();
        let alert = UIAlertController(title: "Checkout Error", message: error, preferredStyle: .alert);
        alert.addAction(UIAlertAction(title: "Ok", style: .default, handler: nil));
        self.present(alert, animated: true);
    }
    
    func checkoutSuccessful(_ viewController: QuadPayCheckoutViewController, orderId: String) {
        viewController.dismiss();
        let alert = UIAlertController(title: "Checkout Successful", message: orderId, preferredStyle: .alert);
        alert.addAction(UIAlertAction(title: "Ok", style: .default, handler: nil));
        self.present(alert, animated: true);
        // Submit order id to your server here!
        //
        //
    }
    
    func checkoutCancelled(_ viewController: QuadPayCheckoutViewController, reason: String) {
        viewController.dismiss();
        let alert = UIAlertController(title: "Checkout Cancelled", message: reason, preferredStyle: .alert);
        alert.addAction(UIAlertAction(title: "Ok", style: .default, handler: nil));
        self.present(alert, animated: true);
    }
    

    @IBAction func buttonTouch(_ sender: Any) {
        NSLog("Button touched");
        let details = QuadPayCheckoutDetails();
        details.amount = NSDecimalNumber(string: "123.54", locale: nil);
        details.customerPhoneNumber = "+11231231234";
        details.merchantReference = "qptest-234-1234";
        details.customerFirstName = "Test";
        details.customerLastName = "Client";
        details.customerAddressLine1 = "100 Any Street";
        details.customerAddressLine2 = "1st";
        details.customerCity = "New York";
        details.customerState = "NY";
        details.customerPostalCode = "10019";
        details.customerCountry = "US";
        let vc = QuadPayCheckoutViewController.start(delegate: self, details: details);
        self.present(vc, animated: true);
    }

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
       
    }


}

